src/redux/reducers/clickReducer.js

```diff
@@ -0,0 +1,46 @@
+import {
+  FETCH_SELECTED_POKEMON,
+  FETCH_SELECTED_ERROR,
+  EXISTING_POKEMON,
+} from "../actions/clickAction";
+
+const initialState = {
+  descriptions: [],
+  isSelected: false,
+  actualPokemon: "",
+  indexActual: 0,
+  description: [],
+  genre: [],
+};
+
+function selectedPokemon(state = initialState, action) {
+  switch (action.type) {
+    case FETCH_SELECTED_POKEMON:
+      return {
+        ...state,
+        indexActual: state.descriptions.length,
+        descriptions: [...state.descriptions, action.payload.description],
+        isSelected: true,
+        actualPokemon: action.payload.name,
+        description: [...state.description, action.payload.text],
+        genre: [...state.genre, action.payload.genre > 4 ? "Female" : "Male"],
```

Comment:
why are you using arrays to store the description and gender, is it for the current pokemon?

Reply:
It is for store all the descriptions and genders that are fetching so far.